src/codegen/mod.rs

```diff
@@ -3871,4 +3918,26 @@ mod utils {
 
         args
     }
+
+    pub fn fnsig_block(
+        ctx: &BindgenContext,
+        sig: &FunctionSig,
+    ) -> quote::Tokens {
+        let args = sig.argument_types().iter().map(|&(_, ty)| {
+            let arg_item = ctx.resolve_item(ty);
+
+            arg_item.to_rust_ty_or_opaque(ctx, &())
+        });
+
+        let return_item = ctx.resolve_item(sig.return_type());
+        let ret_ty = if let TypeKind::Void = *return_item.kind().expect_type().kind() {
+            quote! { () }
+        } else {
+            return_item.to_rust_ty_or_opaque(ctx, &())
+        };
+
+        quote! {
+            *const ::block::Block<(#(#args),*), #ret_ty>
```

Comment:
Late to the party, but if there is a single arg then this expands to `Block<(arg), ret>` which parses the same as `Block<arg, ret>`. What's needed here is `Block<(arg,), ret>` so that args is a single-element tuple.

Reply:
agree, this should be a bug, I will submit a PR for this, thanks very much!